### ISSUE TITLE
The Mobile Launcher window doesn't find Android devices with hyphens in their name

### ISSUE THREAD

**Issue by jamiebrynes7** [spatialos/gdk-for-unity#1288]
**Affects:** Release v0.3.0 and up
**Internal Ticket:** [UTY-2455](https://improbableio.atlassian.net/browse/UTY-2455)

---

#### Description

If your Android device has a hyphen(`-`) in the product name, the mobile launcher will not display it as an option to launch to.

You can see the product name by running the following command in any directory with your Android device connected to your machine:

```
$ adb devices -l
```

#### Workaround

Fork the mobile package and fix the regex:

1. Copy the `io.improbable.gdk.mobile` directory from https://github.com/spatialos/gdk-for-unity/tree/0.3.3/workers/unity/Packages into your Unity project's `Packages` directory.
2. Find the `DeviceMatchRegex` in the `AndroidUtils` static class (https://github.com/spatialos/gdk-for-unity/blob/0.3.3/workers/unity/Packages/io.improbable.gdk.mobile/Editor/AndroidUtils.cs#L28)
3. Replace the entire regex with:

```csharp
    private static readonly Regex DeviceMatchRegex = new Regex(pattern:
        "(?:(?<id>[\\w\\d_-]+)\\s*device).*" +
        "(?:product:(?<product>[\\w\\d_-]+))\\s*" +
        "(?:model:(?<model>[\\w\\d_-]+))\\s*" +
        "(?:device:(?<device>[\\w\\d_-]+)).*");
```

**Comment by paulbalaji** [comment]
A fix has been merged (#1296) and will be rolled out with our next release.

**Comment by jamiebrynes7** [comment]
Fixed in [0.3.4](https://github.com/spatialos/gdk-for-unity/releases/tag/0.3.4)
